src/software/geom/algorithms/step_along_perimeter.cpp

```diff
@@ -0,0 +1,75 @@
+#include "software/geom/algorithms/step_along_perimeter.h"
+
+#include <algorithm>
+#include <vector>
+
+#include "software/geom/algorithms/collinear.h"
+#include "software/geom/algorithms/distance.h"
+#include "software/geom/segment.h"
+
+
+Point stepAlongPerimeter(const Polygon& polygon, const Point& start,
+                         double travel_distance)
+{
+    if (travel_distance == 0.0)
+    {
+        return start;
+    }
+
+    std::vector<Segment> polygon_segments = polygon.getSegments();
```

Comment:
nit: We can get a const reference instead here
```suggestion
    const std::vector<Segment>& polygon_segments = polygon.getSegments();
```